src/node_report_module.cc

```diff
@@ -0,0 +1,506 @@
+  // set up watchdog thread and handler
+  if ((nodereport_events & NR_SIGNAL) && (signal_thread_initialised == false)) {
+    SetupSignalHandler();
+  }
+  // If report no longer required on external user signal,
+  // reset the OS signal handler
+  if (!(nodereport_events & NR_SIGNAL) && (previous_events & NR_SIGNAL)) {
+    RestoreSignalHandler(nodereport_signal, &saved_sa);
+  }
+#endif
+}
+void SetSignal(const FunctionCallbackInfo<Value>& info) {
+#ifndef _WIN32
+  String::Utf8Value parameter(info.GetIsolate(), info[0]);
+  unsigned int previous_signal = nodereport_signal;  // save previous setting
+  nodereport_signal = ProcessNodeReportSignal(*parameter);
+
+  // If signal event active and selected signal has changed,
+  // switch the OS signal handler
+  if ((nodereport_events & NR_SIGNAL) &&
+      (nodereport_signal != previous_signal)) {
+    RestoreSignalHandler(previous_signal, &saved_sa);
+    RegisterSignalHandler(nodereport_signal, SignalDump, &saved_sa);
+  }
+#endif
+}
+void SetFileName(const FunctionCallbackInfo<Value>& info) {
+  String::Utf8Value parameter(info.GetIsolate(), info[0]);
+  ProcessNodeReportFileName(*parameter);
+}
+void SetDirectory(const FunctionCallbackInfo<Value>& info) {
+  String::Utf8Value parameter(info.GetIsolate(), info[0]);
+  ProcessNodeReportDirectory(*parameter);
+}
+void SetVerbose(const FunctionCallbackInfo<Value>& info) {
+  String::Utf8Value parameter(info.GetIsolate(), info[0]);
+  nodereport_verbose = ProcessNodeReportVerboseSwitch(*parameter);
+}
+
+/*******************************************************************************
+ * Callbacks for triggering report on fatal error, uncaught exception and
+ * external signals
+ ******************************************************************************/
+static void OnFatalError(const char* location, const char* message) {
+  if (location) {
+    fprintf(stderr, "FATAL ERROR: %s %s\n", location, message);
+  } else {
+    fprintf(stderr, "FATAL ERROR: %s\n", message);
+  }
+  // Trigger report if requested
+  if (nodereport_events & NR_FATALERROR) {
+    TriggerNodeReport(Isolate::GetCurrent(),
+                      kFatalError,
+                      message,
+                      location,
+                      nullptr,
+                      MaybeLocal<Value>());
+  }
+  fflush(stderr);
+  raise(SIGABRT);
+}
+
+bool OnUncaughtException(Isolate* isolate) {
+  // Trigger report if requested
+  if (nodereport_events & NR_EXCEPTION) {
+    TriggerNodeReport(isolate,
+                      kException,
+                      "exception",
+                      __func__,
+                      nullptr,
+                      MaybeLocal<Value>());
+  }
+  if ((version_and_command.commandline_string.find(
+       "abort-on-uncaught-exception")
+       != std::string::npos) ||
+      (version_and_command.commandline_string.find(
+       "abort_on_uncaught_exception")
+       != std::string::npos)) {
```

Comment:
style nit: 4 spaces indentation for statement continuations

Reply:
done